lib/Driver/ToolChains.cpp

```diff
@@ -85,6 +85,24 @@ static void addPrimaryInputsOfType(ArgStringList &Arguments,
   }
 }
 
+static bool
+addOutputsOfType(ArgStringList &Arguments,
+                 CommandOutput const &Output,
+                 const llvm::opt::ArgList &Args,
+                 types::ID OutputType,
+                 const char *PrefixArgument = nullptr) {
+  bool Added = false;
+  for (auto Output : Output.getAdditionalOutputsForType(OutputType)) {
+    if (Output.empty())
+      continue;
```

Comment:
Okay, I take it back, this doesn't make any sense. We're building a new vector and the vector has empty strings in it? That really *shouldn't* happen.

Reply:
well .. yeah, the empty strings only arise because "additional outputs" and "primary outputs" can intersect -- by type -- and primary outputs can be empty (though I'd like it if that wasn't true).

Reply:
I'm not sure how primary outputs would be empty at a time when additional outputs *wouldn't* be, or why primary outputs would end up in the result of `getAdditionalOutputsForType`.

Reply:
(I can assert that they're disjoint and always make "additional output of type primary-type" === "primary output", but then multi-threading.swift still fails because (for reasons I'm not clear on) it adds a dependencies file for main.swift, but not multi-threaded.swift. Possibly we're just not adding enough additional outputs? I think I went down this road debugging before, will try again)

Reply:
Hm maybe I'm thinking of this wrong and the issue is that these jobs have multiple primary inputs but only one primary _output_, and we should have the same number of aux outputs as primary outputs? Hmm. no that doesn't work either; multi-threaded.swift has 2 object outputs, one for each primary input. Sigh.

Reply:
Why would it matter that the dependencies file is only there for main.swift and not multi-threaded.swift? You wouldn't ask for an auxiliary file by input name in WMO.

Reply:
Number of primary outputs should always match number of primary inputs. Number of auxiliary outputs depends on what kind of job it is, but it's either 0, 1, or N. If that's all we can assert right now, then okay.

Reply:
We really, really shouldn't have empty strings, though. That will actually break things.